event-streams/src/event-streams-v1.0-wd01.md

```diff
@@ -0,0 +1,547 @@
+              | <expression> <comparison-operator> <expression>
+              | <predicate> <binary-logical-operator> <predicate>
+expression ::= <constant> | <field_value>  
+constant   ::= <string_constant> | <integer_constant>
+field_value := <field>
+field ::= <section> ’.’<field_name>
+section ::= { ‘message_annotations’ | ‘m’ }
+```
+
+This grammar subset allows for a minimal implementation that is limited to
+evaluating the delivery annotation properties defined here while being
+interoperable with the full SQL filter syntax.
+
+Examples:
+
+- Start from event at offset "a4c5" exclusively: `m.event-streams-offset > '100'`
+- Start from event at offset "a4c5" inclusively: `m.event-streams-offset >= '100'`
+- Start from event received after the timestamp: `m.event-streams-timestamp > 1585672841`
+
+
+``` XML
+ <type name="event-streams-sql-filter" class="restricted" source="sql-filter" provides="filter">
+<descriptor name="amqp:event-streams-sql-filter" code="0x00000000:0x00000201"/> 
+</type>
+```
+
+The filter-type is “amqp:event-streams-sql-filter”, with type-code 0x00000000:0x00000201
+
+
+## 2.3 Link ownership
+
+An implementation MAY constrain the number of concurrent receivers that can
+receive from a log or its partitions.
+
+In scenarios where a group of multiple concurrent event processors want to
+receive events from a partitioned event log and where they use an external
+consensus mechanism to agree on which event processor owns which partition, it
+is desireable to restrict each partition to one (the owning) receiver from the
+group and for event processors to "steal" ownership of the partition as
+assignments amongst the group change.
+
+This specification does not prescribe any algorithm or strategy for reaching
+such consensus, but only a mechanism for realizing its outcome:
+
+For an event log that is partitioned, a link MAY be associated with a partition
+and a consumer group when the respective link properties are provided during
+`attach`. For the combination of partition and consumer group, the node
+maintains a `long`-valued `epoch` counter, which is seeded with zero and
+returned in the `event-streams-epoch` link property to the receiver during a
+successful attempt to attach.
+
+Any subsequent attempt to attach a link with the same combination of partition
+and consumer group link property values will be rejected while there is an
+active link and unless the newly attaching receiver provides an epoch counter
+greater than the current epoch counter value using the `event-streams-epoch`
+link property. If that condition is true, the new epoch counter value is
+recorded, any existing link for the given partition and consumer group is closed
+by the node, and the new link is attached.
+
+Whenever the link for a partition within a group is closed by the receiver and
+therefore no links remain, the respective epoch counter is reset to zero.  
+
+### 2.2.1 event-streams-consumer-group link property
```

Comment:
Would this property be also used to indicate the consumer group when the event-log node is coordinating the group membership and partition distribution on its own? Ot is tht expected to use different mechanism - e.g. in the address or something?

Reply:
I moved things around to allow for that.